src/xmppjs/XJSAccount.ts

```diff
@@ -186,18 +193,22 @@ export class XmppJsAccount implements IBifrostAccount {
     }
 
     public async joinChat(
-        components: IChatJoinProperties,
+        components: { handle: string }&({room: string, server: string}|{fullRoomName: string}),
         instance?: IBifrostInstance,
         timeout: number = 5000,
         setWaiting: boolean = true)
         : Promise<IConversationEvent|void> {
-        if (!components.fullRoomName && (!components.room || !components.server)) {
-            throw Error("Missing fullRoomName OR room|server");
-        }
+        let roomName: string;
         if (!components.handle) {
             throw Error("Missing handle");
```

Comment:
Can this ever happen with the typecheck in place?  Or do some callers pass an `any` to this? (Or can `handle` ever be an empty string?)

Reply:
We had it before where this happened and it blew up :(